createSpotifyPlaylist.py

```diff
@@ -0,0 +1,217 @@
+        ----------
+        spotify_playlist_id : str
+            The uri (id) of a spotify playlist
+
+        Returns
+        -------
+        None
+
+        """
+        # get spotify playlist
+        playlist_tracks = self.spotify_client.playlist(playlist_id=spotify_playlist_id)
+        # select the first tracks popularity number
+        most_popular = playlist_tracks['tracks']['items'][0]['track']['popularity']
+
+        # search for most popular tracks
+        for track in playlist_tracks['tracks']['items']:
+            if track['track'] is None:
+                pass
+            else:
+                popularity = track['track']['popularity'] # get popularity number
+                uri = track['track']['uri'] # get uri for the track
+                # add popular tracks to list of popularity
+                if uri not in self.popular_tracks_list and popularity > most_popular:
+                    self.popular_tracks_list.append(uri)
+        return 0
+
+
+
+    def create_spotify_playlist(self, playlist_name:str, for_user:str):
+        """Functio to create spotify playlist
+
+        This function will create a new spotify playlist with the title provied by 'playlist_name'
+        and `for_user`. Once we have created the new spotify playlist we return the playlist id for
+        later use. This function has no returns
+
+        Parameters
+        ----------
+        playlist_name : str
+            The name of the new spotify playlist.
+        for_user : str
+            The name of the user who requested a playlist.
+
+        Returns
+        -------
+        str
+            A string containg the new spotify playlist id.
+
+        """
+
+        playlist_name = f'{playlist_name} for user {for_user}'
+
+        new_playlist = self.spotify_client.user_playlist_create(
+            user=os.environ['spotifyUserID'],
+            name=playlist_name, public=True,
+            collaborative=False,
+            description='a bot created this'
+            )
+        return new_playlist['id']
+
+
+
+    def add_tracks_to_playlist(self, playlist_id:str):
+        """Functio to add songs to a spotify playlist
+
+        This function will addd tracks to a playlist. This function has no returns
+
+        Parameters
+        ----------
+        playlist_id : str
+            The new spotify playlist id
+
+        Returns
+        -------
+        None
+
+        """
+        list_one = self.popular_tracks_list[:len(self.popular_tracks_list)//2]
+        list_two = self.popular_tracks_list[len(self.popular_tracks_list)//2:]
+
+        self.spotify_client.playlist_add_items(
+            playlist_id=playlist_id,
+            items=list_one,
+            position=None
+        )
+        self.spotify_client.playlist_add_items(
+            playlist_id=playlist_id,
+            items=list_two,
+            position=None
+        )
+
+        return 0
+
+
+
+def create_spotify_playlist_from_search(query:str, name:str):
```

Comment:
spaces afer colons - also add the type hint for the return

Reply:
fixed spacing issues on the module. check it out [here](https://github.com/GaelGil/apiPractice/commit/8b206b0d7eab905d3c1373ddb1e869ecffa58e70)